ecs_compatibility: revert breaking change; keep `disabled` as default for 8.0 (#13080)
elastic/logstash: docs/static/ecs-compatibility.asciidoc

Any plugin that supports this mode will also have an `ecs_compatibility` option, which allows you to configure which mode the individual plugin instance should operate in.
If left unspecified for an individual plugin, the pipeline's `pipeline.ecs_compatibility` setting will be observed.
This allows you to configure plugins to use ECS -- or to lock in your existing non-ECS behavior.

> ECS Compatibility modes do not prevent you from explicitly configuring a plugin in a manner that conflicts with ECS.

Reviewer: ```suggestion
ECS compatibility modes do not prevent you from explicitly configuring a plugin in a manner that conflicts with ECS.
```

Instead, they ensure that _implicit_ configuration avoids conflicts.

> NOTE: Until {ls} 8.0 and the final 7.x are released, any value for `pipeline.ecs_compatibility` other than `disabled` are considered BETA and unsupported.

Reviewer: ```suggestion
NOTE: Until {ls} 8.0 and the final 7.x are released, any value for `pipeline.ecs_compatibility` other than `disabled` is considered BETA and unsupported.
```

>       As we continue to release plugins with ECS Compatibility modes, having this flag set will cause even patch-level upgrades to _automatically_ consume breaking changes in the upgraded plugins, changing the shape of data the plugin produces.

Reviewer: ```suggestion
      As we continue to release plugins with ECS compatibility modes, having this flag set will cause even patch-level upgrades to _automatically_ consume breaking changes in the upgraded plugins, changing the shape of data the plugin produces.
```

ifeval::["{ls8-ecs-major-version}"!="v8"]

> NOTE: ECS `v8` will be available along-side the GA release of {ls} 8.0.0, and will be available at or before the final minor release of {ls} 7.

Reviewer: ```suggestion
NOTE: ECS `v8` will be available alongside the GA release of {ls} 8.0.0, and will be available at or before the final minor release of {ls} 7.
```

  pipeline.ecs_compatibility: {ls8-ecs-major-version}
-----

> NOTE: Until the General Availability of {ls} 8.0.0 that coincides with the final minor release of {ls} 7, any value for `pipeline.ecs_compatibility` other than `disabled` is considered BETA and unsupported because it will produce undesireable consequences when performing upgrades.

Reviewer: ```suggestion
NOTE: Until the General Availability of {ls} 8.0.0 that coincides with the final minor release of {ls} 7, any value for `pipeline.ecs_compatibility` other than `disabled` is considered BETA and unsupported because it will produce undesirable consequences when performing upgrades.
```
